### ISSUE TITLE
packages contain code that will be rejected by a future version of Rust

### ISSUE THREAD

**Issue by brianmay** [influxdb-rs/influxdb-rust#178]
```
warning: the following packages contain code that will be rejected by a future version of Rust: influxdb v0.7.2
note:
To solve this problem, you can try the following approaches:

- If the issue is not solved by updating the dependencies, a fix has to be
implemented by those dependencies. You can help with that by notifying the
maintainers of this problem (e.g. by creating a bug report) or by proposing a
fix to the maintainers (e.g. by creating a pull request):

  - influxdb@0.7.2
  - Repository: https://github.com/influxdb-rs/influxdb-rust
  - Detailed warning command: `cargo report future-incompatibilities --id 1 --package influxdb@0.7.2`

- If waiting for an upstream fix is not an option, you can use the `[patch]`
section in `Cargo.toml` to use your own version of the dependency. For more
information, see:
https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#the-patch-section

note: this report can be shown with `cargo report future-incompatibilities --id 1`
```

Running that command gives:

```console
$ cargo report future-incompatibilities --id 1
The following warnings were discovered during the build. These warnings are an
indication that the packages contain code that will become an error in a
future release of Rust. These warnings typically cover changes to close
soundness problems, unintended or undocumented behavior, or critical problems
that cannot be fixed in a backwards-compatible fashion, and are not expected
to be in wide use.

Each warning should contain a link for more information on what the warning
means and how to resolve it.


To solve this problem, you can try the following approaches:

- If the issue is not solved by updating the dependencies, a fix has to be
implemented by those dependencies. You can help with that by notifying the
maintainers of this problem (e.g. by creating a bug report) or by proposing a
fix to the maintainers (e.g. by creating a pull request):

  - influxdb@0.7.2
  - Repository: https://github.com/influxdb-rs/influxdb-rust
  - Detailed warning command: `cargo report future-incompatibilities --id 1 --package influxdb@0.7

- If waiting for an upstream fix is not an option, you can use the `[patch]`
section in `Cargo.toml` to use your own version of the dependency. For more
information, see:
https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#the-patch-section

The package `influxdb v0.7.2` currently triggers the following future incompatibility lints:
> warning: cannot find macro `cargo_toml` in the current scope when looking from the crate root
>   --> /home/brian/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs
>    |
> 23 | #![doc = cargo_toml!(indent="", "derive")]
>    |          ^^^^^^^^^^ not found from the crate root
>    |
>    = warning: this was previously accepted by the compiler but is being phased out; it will beco
>    = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124
>    = help: import `macro_rules` with `use` to make it callable above its definition
>
> warning: cannot find macro `cargo_toml_private` in the current scope when looking from the crate
>    --> /home/brian/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.r
>     |
>  23 | #![doc = cargo_toml!(indent="", "derive")]
>     |          -------------------------------- in this macro invocation
> ...
> 104 |         cargo_toml_private!($indent, "", $firstfeat $(, $feature)*)
>     |         ^^^^^^^^^^^^^^^^^^ not found from the crate root
>     |
>     = warning: this was previously accepted by the compiler but is being phased out; it will bec
>     = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/12
>     = help: import `macro_rules` with `use` to make it callable above its definition
>     = note: this warning originates in the macro `cargo_toml` (in Nightly builds, run with -Z ma
>
> warning: cannot find macro `cargo_toml` in the current scope when looking from the crate root
>   --> /home/brian/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs
>    |
> 80 | #![doc = cargo_toml!(indent="\t", "derive")]
>    |          ^^^^^^^^^^ not found from the crate root
>    |
>    = warning: this was previously accepted by the compiler but is being phased out; it will beco
>    = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124
>    = help: import `macro_rules` with `use` to make it callable above its definition
>
> warning: cannot find macro `cargo_toml_private` in the current scope when looking from the crate
>    --> /home/brian/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.r
>     |
>  80 | #![doc = cargo_toml!(indent="\t", "derive")]
>     |          ---------------------------------- in this macro invocation
> ...
> 104 |         cargo_toml_private!($indent, "", $firstfeat $(, $feature)*)
>     |         ^^^^^^^^^^^^^^^^^^ not found from the crate root
>     |
>     = warning: this was previously accepted by the compiler but is being phased out; it will bec
>     = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/12
>     = help: import `macro_rules` with `use` to make it callable above its definition
>     = note: this warning originates in the macro `cargo_toml` (in Nightly builds, run with -Z ma
>
> warning: cannot find macro `cargo_toml` in the current scope when looking from the crate root
>   --> /home/brian/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs
>    |
> 82 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "reqwest
>    |          ^^^^^^^^^^ not found from the crate root
>    |
>    = warning: this was previously accepted by the compiler but is being phased out; it will beco
>    = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124
>    = help: import `macro_rules` with `use` to make it callable above its definition
>
> warning: cannot find macro `cargo_toml_private` in the current scope when looking from the crate
>    --> /home/brian/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.r
>     |
>  82 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "reqwes
>     |          ---------------------------------------------------------------------------------
> ...
> 108 |         cargo_toml_private!($indent, "default-features = false,", $firstfeat $(, $feature)
>     |         ^^^^^^^^^^^^^^^^^^ not found from the crate root
>     |
>     = warning: this was previously accepted by the compiler but is being phased out; it will bec
>     = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/12
>     = help: import `macro_rules` with `use` to make it callable above its definition
>     = note: this warning originates in the macro `cargo_toml` (in Nightly builds, run with -Z ma
>
> warning: cannot find macro `cargo_toml` in the current scope when looking from the crate root
>   --> /home/brian/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs
>    |
> 84 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "hyper-c
>    |          ^^^^^^^^^^ not found from the crate root
>    |
>    = warning: this was previously accepted by the compiler but is being phased out; it will beco
>    = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124
>    = help: import `macro_rules` with `use` to make it callable above its definition
>
> warning: cannot find macro `cargo_toml_private` in the current scope when looking from the crate
>    --> /home/brian/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.r
>     |
>  84 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "hyper-
>     |          ---------------------------------------------------------------------------------
> ...
> 108 |         cargo_toml_private!($indent, "default-features = false,", $firstfeat $(, $feature)
>     |         ^^^^^^^^^^^^^^^^^^ not found from the crate root
>     |
>     = warning: this was previously accepted by the compiler but is being phased out; it will bec
>     = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/12
>     = help: import `macro_rules` with `use` to make it callable above its definition
>     = note: this warning originates in the macro `cargo_toml` (in Nightly builds, run with -Z ma
>
> warning: cannot find macro `cargo_toml` in the current scope when looking from the crate root
>   --> /home/brian/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs
>    |
> 86 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "curl-cl
>    |          ^^^^^^^^^^ not found from the crate root
>    |
>    = warning: this was previously accepted by the compiler but is being phased out; it will beco
>    = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124
>    = help: import `macro_rules` with `use` to make it callable above its definition
>
> warning: cannot find macro `cargo_toml_private` in the current scope when looking from the crate
>    --> /home/brian/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.r
>     |
>  86 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "curl-c
>     |          ---------------------------------------------------------------------------------
> ...
> 108 |         cargo_toml_private!($indent, "default-features = false,", $firstfeat $(, $feature)
>     |         ^^^^^^^^^^^^^^^^^^ not found from the crate root
>     |
>     = warning: this was previously accepted by the compiler but is being phased out; it will bec
>     = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/12
>     = help: import `macro_rules` with `use` to make it callable above its definition
>     = note: this warning originates in the macro `cargo_toml` (in Nightly builds, run with -Z ma
>
> warning: cannot find macro `cargo_toml` in the current scope when looking from the crate root
>   --> /home/brian/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs
>    |
> 88 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "h1-clie
>    |          ^^^^^^^^^^ not found from the crate root
>    |
>    = warning: this was previously accepted by the compiler but is being phased out; it will beco
>    = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124
>    = help: import `macro_rules` with `use` to make it callable above its definition
>
> warning: cannot find macro `cargo_toml_private` in the current scope when looking from the crate
>    --> /home/brian/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.r
>     |
>  88 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "h1-cli
>     |          ---------------------------------------------------------------------------------
> ...
> 108 |         cargo_toml_private!($indent, "default-features = false,", $firstfeat $(, $feature)
>     |         ^^^^^^^^^^^^^^^^^^ not found from the crate root
>     |
>     = warning: this was previously accepted by the compiler but is being phased out; it will bec
>     = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/12
>     = help: import `macro_rules` with `use` to make it callable above its definition
>     = note: this warning originates in the macro `cargo_toml` (in Nightly builds, run with -Z ma
>
> warning: cannot find macro `cargo_toml` in the current scope when looking from the crate root
>   --> /home/brian/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs
>    |
> 90 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "h1-clie
>    |          ^^^^^^^^^^ not found from the crate root
>    |
>    = warning: this was previously accepted by the compiler but is being phased out; it will beco
>    = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124
>    = help: import `macro_rules` with `use` to make it callable above its definition
>
> warning: cannot find macro `cargo_toml_private` in the current scope when looking from the crate
>    --> /home/brian/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.r
>     |
>  90 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "h1-cli
>     |          ---------------------------------------------------------------------------------
> ...
> 108 |         cargo_toml_private!($indent, "default-features = false,", $firstfeat $(, $feature)*)
>     |         ^^^^^^^^^^^^^^^^^^ not found from the crate root
>     |
>     = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>     = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124535>
>     = help: import `macro_rules` with `use` to make it callable above its definition
>     = note: this warning originates in the macro `cargo_toml` (in Nightly builds, run with -Z macro-backtrace for more info)
>
> warning: cannot find macro `cargo_toml` in the current scope when looking from the crate root
>   --> /home/brian/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs:92:10
>    |
> 92 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "wasm-client")]
>    |          ^^^^^^^^^^ not found from the crate root
>    |
>    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>    = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124535>
>    = help: import `macro_rules` with `use` to make it callable above its definition
>
> warning: cannot find macro `cargo_toml_private` in the current scope when looking from the crate root
>    --> /home/brian/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs:108:9
>     |
>  92 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "wasm-client")]
>     |          ---------------------------------------------------------------------------------------- in this macro invocation
> ...
> 108 |         cargo_toml_private!($indent, "default-features = false,", $firstfeat $(, $feature)*)
>     |         ^^^^^^^^^^^^^^^^^^ not found from the crate root
>     |
>     = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>     = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124535>
>     = help: import `macro_rules` with `use` to make it callable above its definition
>     = note: this warning originates in the macro `cargo_toml` (in Nightly builds, run with -Z macro-backtrace for more info)
>
```

-   Rust Version: 1.91.1
-   Runtime version (e.g. Tokio):
-   OS Version: NixOs 25.05

Steps to Reproduce:

1. `cargo clippy --future-incompat-report`
